src/pages/_app.tsx

```diff
@@ -52,17 +53,19 @@ export default function MyApp(props: MyAppProps) {
             <AccountProvider>
               <CoretimeApiContextProvider>
                 <RelayApiContextProvider>
-                  <ContextDataProvider>
-                    <TaskDataProvider>
-                      <RegionDataProvider>
-                        <MarketProvider>
-                          <SaleInfoProvider>
-                            {getLayout(<Component {...pageProps} />)}
-                          </SaleInfoProvider>
-                        </MarketProvider>
-                      </RegionDataProvider>
-                    </TaskDataProvider>
-                  </ContextDataProvider>
+                  <BalanceProvider>
+                    <ContextDataProvider>
+                      <TaskDataProvider>
+                        <RegionDataProvider>
+                          <MarketProvider>
+                            <SaleInfoProvider>
+                              {getLayout(<Component {...pageProps} />)}
+                            </SaleInfoProvider>
+                          </MarketProvider>
+                        </RegionDataProvider>
```

Comment:
We are starting to have a lot of context providers nested. Not sure how bad this is.